src/Transit/Internal/Pipeline.hs

```diff
@@ -67,37 +66,40 @@ encryptC key = go Saltine.zero
               cipherTextSize = toLazyByteString (word32BE (fromIntegral (BS.length cipherText)))
           C.yield (toS cipherTextSize)
           C.yield cipherText
-          go (Saltine.nudge nonce)
+          loop (Saltine.nudge nonce)
 
 decryptC :: MonadIO m => SecretBox.Key -> C.ConduitT ByteString ByteString m ()
-decryptC key = loop
+decryptC key = loop Saltine.zero
   where
-    loop = do
+    loop :: MonadIO m => SecretBox.Nonce -> C.ConduitT ByteString ByteString m ()
+    loop seqNum = do
       b <- C.await
       case b of
         Nothing -> return ()
         Just bs -> do
-          let (nonceBytes, ciphertext) = BS.splitAt boxNonce bs
-              nonce = fromMaybe (panic "unable to decode nonce") $
-                Saltine.decode nonceBytes
-              maybePlainText = SecretBox.secretboxOpen key nonce ciphertext
-          case maybePlainText of
-            Just plaintext -> do
-              C.yield plaintext
-              loop
-            Nothing -> throwIO (CouldNotDecrypt "SecretBox failed to open")
+          case decrypt key bs of
+            Right (plainText, nonce) -> do
+              let seqNumLE = BS.reverse $ toS $ Saltine.encode seqNum
```

Comment:
Yes. Naming is just too hard (for me). :-( Tried to avoid creating that intermediate identifier, however, the next expression grew too long and unreadable.